lib/llvmopencl/WorkItemAliasAnalysis.cc

```diff
@@ -116,9 +116,19 @@ WorkItemAAResult WorkItemAA::run(Function &F, AnalysisManager<Function> *AM) {
     return WorkItemAAResult(AM->getResult<WorkItemAA>(F));
 }
 
-bool WorkItemAliasAnalysis::runOnFunction(llvm::Function &) {
+bool WorkItemAliasAnalysis::runOnFunction(
+    #ifndef LLVM_10_0
+    [gcc::unused]
```

Comment:
This looks weird, I'd expect it to warn in case not detecting the attribute, but it looks like it doesn't detect attributes in general (introduced in C++11, I think, and C++11 has been required for a while for LLVM builds).